src/main/java/org/jenkinsci/plugins/orgfolder/github/GitHubLink.java

```diff
@@ -29,6 +32,10 @@
         this(image,url.toExternalForm());
     }
 
+    private Object readResolve() throws ObjectStreamException {
+        return new org.jenkinsci.plugins.github_branch_source.GitHubLink("icon-github-"+image, url);
```

Comment:
Note that you could also use XStream aliases and completely delete classes like this, if the fields are compatible.